### ISSUE TITLE
Support deployment to active K8s context

### ISSUE THREAD

**Issue by ivanporty** [GoogleCloudPlatform/cloud-code-intellij#2517]
With https://github.com/GoogleCloudPlatform/cloud-code-intellij-internal/pull/90 is place, the context setting for a run configuration is now mandatory, and it means the run configuration always deploys to a specific context.

We probably still need to support deploying to whatever active cluster is, without hard binding to a specific one (when people change clusters outside of IDEs), and make it a default choice.

@matthewmichihara proposed empty context combobox selection for active default one.

Another option:
additional fake item with default/active cluster, with its name in parentheses. I.e.

* Active [current: minikube]
* docker-for-desktop
* GKE-some-cluster
* minikube

The only problem is we kind of list active cluster once and then list it separately again, need to make sure this is not confusing...

**Comment by matthewmichihara** [comment]
One additional option we can think about is potentially adding another checkbox that toggles the enabled state of the context chooser, like this example from the Application configuration:

![image](https://user-images.githubusercontent.com/79073/56690775-49cb9e80-66ac-11e9-889b-cd339f4cd024.png)

But instead of saying `Redirect input from`, it says `Pin to Kubernetes context`, and only when checked does the combobox to the right of it become active.

We also need to consider how this interacts with https://github.com/GoogleCloudPlatform/cloud-code-intellij/issues/2515

**Comment by etanshaul** [comment]
Could you help me understand what the scenarios are where there won't be a context that we want (or can) pin to?

**Comment by matthewmichihara** [comment]
@ivanporty correct me if I'm wrong but the scenario we're trying to support here is where a user wants to manage contexts via `kubectl` and have the intellij run configs just deploy to whatever the current cluster is. Whether this is a common use case is up for debate.

**Comment by etanshaul** [comment]
If the user changes the context via kubectl, then this changes context in kubeconfig (and hence will update in our UI as well). I may be misunderstanding the scenario though.

**Comment by matthewmichihara** [comment]
If the user changes the current context via kubectl, it does change the current context in kubeconfig, but that won't change any pinned contexts they have for existing run configurations.

Currently, I have 2 contexts -- `docker` and `gke`. The current context is `gke`. I have an existing run configuration pinned to `gke`. If I `kubectl config use-context docker`, it changes my system context, but the run configuration is still pinned to `gke`.

Does that match what you expected?

**Comment by etanshaul** [comment]
Oh I see!

Yeah. My very initial intuition here is that we shouldn't change the selected context in existing run configurations. If the user makes a selection in the run config, it should be honored indefinitely.

That being said, having some clever way of changing this behavior is worth considering (but the default behavior imo is to honor the selection). Definitely open to debate though.

**Comment by ivanporty** [comment]
Imagine people working on Mac: they sort of used to switching contexts with Docker menu, some may use Cluster explorer for that once it's done, etc. There are just  too many ways to switch contexts and we may want to keep the option to deploy to whatever active one is.

**Comment by etanshaul** [comment]
> Imagine people working on Mac: they sort of used to switching contexts with Docker menu, some may use Cluster explorer for that once it's done, etc. There are just too many ways to switch contexts and we may want to keep the option to deploy to whatever active one is.

yes, totally agree. I'm just suggesting that the _default_ option be to pin a run config to a context indefinitely. Then we can consider how to expose the option to use the system context in the UI in a clever way (perhaps using a checkbox like @matthewmichihara suggested). I think we are in agreement?

**Comment by ivanporty** [comment]
Yes, sounds perfect. I agree, checkbox option look pretty good!

**Comment by ivanporty** [comment]
Assuming it will be checked by default, so we pin the configuration to a selected context, and a developer would need to uncheck it to always deploy to active one.

**Comment by matthewmichihara** [comment]
SGTM. Yeah having it checked by default sounds like the right move.

**Comment by etanshaul** [comment]
@matthewmichihara can this be closed?

**Comment by matthewmichihara** [comment]
I don't think so since we currently don't support deployment to the active kubernetes context. Once you view the run configuration, it has to get pinned to some context.

**Comment by etanshaul** [comment]
oh thats right. I misread this issue. nm. this is lower priority anyway, but we'll leave open to track
